Automattic/woocommerce-payments
# Remove unsupported WC Pay currencies from the available currencies array

Fixes #2490

#### Changes proposed in this Pull Request

- Add a filter on the available currencies for additional currencies to be added by other payment gateways, custom code, or anything else
- Remove unsupported WC Pay currencies from the available currencies array

Even though this PR depends on 992-gh-Automattic/woocommerce-payments-server to get the presentment currencies list, it should work without it. So, we can merge it without having to merge the server PR first.










#### Testing instructions





* Navigate to the Multi-Currency settings page and click on the Add currencies button
* Assert that all WooCommerce currencies are available in the modal (BTC should be included)
* Apply 992-gh-Automattic/woocommerce-payments-server to your local server or sandbox
* Clear the account cache using the dev plugin and make sure the `presentment_currencies` array is there
* Navigate to the Multi-Currency settings page and click on the Add currencies button
* Assert that only `presentment_currencies` that are also part of the WooCommerce currencies list are available in the modal (BTC should not show up)
* Use the `wcpay_multi_currency_available_currencies` filter to add BTC to the currencies list
* Navigate to the Multi-Currency settings page and click on the Add currencies button
* Assert that BTC is available the modal

-------------------
